eng/common/scripts/Verify-RestApiSpecLocation.ps1

```diff
@@ -0,0 +1,281 @@
+                foreach($inputFile in $inputFileValue) {
+                  Verify-Url $inputFile
+                }
+              }
+            }
+          }
+        }
+      }
+      catch {
+        Write-Host "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Failed to parse yaml section $yamlSection with exception:`n$_ "
+      }
+    }
+  }
+}
+
+function Verify-PackageVersion() {
+  try {
+    $packages = @{}
+    if ($FindArtifactForApiReviewFn -and (Test-Path "Function:$FindArtifactForApiReviewFn"))
+    {
+      $packages = &$FindArtifactForApiReviewFn $ArtifactLocation $PackageName
+    }
+    else
+    {
+      LogError "The function for 'FindArtifactForApiReviewFn' was not found.`
+      Make sure it is present in eng/scripts/Language-Settings.ps1 and referenced in eng/common/scripts/common.ps1.`
+      See https://github.com/Azure/azure-sdk-tools/blob/main/doc/common/common_engsys.md#code-structure"
+      exit 1
+    }
+    if (-not $PackageInfoDirectory)
+    {
+      $PackageInfoDirectory = Join-Path -Path $ArtifactLocation "PackageInfo"
+      if (-not (Test-Path -Path $PackageInfoDirectory)) {
+        # Call Save-Package-Properties.ps1 script to generate package info json files
+        $savePropertiesScriptPath = Join-Path -Path $PSScriptRoot "Save-Package-Properties.ps1"
+        & $savePropertiesScriptPath -serviceDirectory $ServiceDirectory -outDirectory $PackageInfoDirectory
+      }
+    }
+
+    $continueValidation = $false
+    if ($packages)
+    {
+      foreach($pkgPath in $packages.Values)
+      {
+        $pkgPropPath = Join-Path -Path $PackageInfoDirectory "$PackageName.json"
+        if (-Not (Test-Path $pkgPropPath))
+        {
+            Write-Host "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Package property file path $($pkgPropPath) is invalid."
+            continue
+        }
+        # Get package info from json file
+        $pkgInfo = Get-Content $pkgPropPath | ConvertFrom-Json
+        $version = [AzureEngSemanticVersion]::ParseVersionString($pkgInfo.Version)
+        if ($null -eq $version)
+        {
+            LogError "ServiceDir:$ServiceDirectory, Version info is not available for package $PackageName, because version '$(pkgInfo.Version)' is invalid. Please check if the version follows Azure SDK package versioning guidelines."
+            exit 1
+        }
+
+        Write-Host "Version: $($version)"
+        Write-Host "SDK Type: $($pkgInfo.SdkType)"
+        Write-Host "Release Status: $($pkgInfo.ReleaseStatus)"
+
+        # Ignore the validation if the package is not GA version
+        if ($version.IsPrerelease) {
+          Write-Host "ServiceDir:$ServiceDirectory, Package $($parsedPackage.PackageId) is marked with version $($parsedPackage.PackageVersion), the version is a prerelease version and the validation of spec location is ignored."
+          exit 0
```

Comment:
Do you want to exit the entire script here? What if there are multiple packages?

Reply:
Nevermind I just looked closer at the code and there shouldn't ever be more than one package returned. I know I told you to use this same method and I do still agree with that but I may want to clean that code up at some point.